superset/migrations/versions/2023-03-27_12-30_7e67aecbf3f1_chart_ds_constraint.py

```diff
@@ -63,9 +66,15 @@ def upgrade():
     session = db.Session(bind=bind)
 
     with op.batch_alter_table("slices") as batch_op:
-        for slc in session.query(Slice).filter(Slice.datasource_type == "query").all():
-            upgrade_slc(slc)
-            session.add(slc)
+        for slc in session.query(Slice).filter(Slice.datasource_type != "table").all():
+            if slc.datasource_type == "query":
+                upgrade_slc(slc)
+                session.add(slc)
+
+            else:
+                logger.warn(
+                    f"unknown value detected for slc.datasource_type: {slc.datasource_type}"
```

Comment:
I would suggest also putting a log into line 60 above in case that's where the failure is.

Reply:
so that's only going to be updating slices that are "query", i updated the session.query to look at all instances that are != table